lib/sidekiq/job_logger.rb

```diff
@@ -3,44 +3,48 @@
 module Sidekiq
   class JobLogger
 
+    def initialize(logger=Sidekiq.logger)
+      @logger = logger
+    end
+
     def call(item, queue)
       start = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
-
-      Sidekiq.logger.info("start")
+      @logger.info("start")
 
       yield
 
       with_elapsed_time_context(start) do
-        Sidekiq.logger.info("done")
+        @logger.info("done")
       end
     rescue Exception
       with_elapsed_time_context(start) do
-        Sidekiq.logger.info("fail")
+        @logger.info("fail")
       end
 
       raise
     end
 
     def with_job_hash_context(job_hash, &block)
-      Sidekiq.logger.with_context(job_hash_context(job_hash), &block)
+      @logger.with_context(job_hash_context(job_hash), &block)
     end
 
     def job_hash_context(job_hash)
       # If we're using a wrapper class, like ActiveJob, use the "wrapped"
       # attribute to expose the underlying thing.
-      {
+      h = {
         class: job_hash['wrapped'] || job_hash["class"],
         jid: job_hash['jid'],
-        bid: job_hash['bid']
       }
+      h[:bid] = job_hash['bid'] if job_hash['bid']
```

Comment:
You don't have to do this, format context code handles `nil` values with `compact` [here](https://github.com/mperham/sidekiq/blob/6-0/lib/sidekiq/logger.rb#L44).